File: 965-unique-email-addresses/unique-email-addresses.js
/**
 * @param {string[]} emails
 * @return {number}
 */
var numUniqueEmails = function(emails) {
    let sett = new Set()
    for(let email of emails){
        let s = email.split("@")
        let local = s[0].split("+")[0].replaceAll(".","")
        let domain = s[1]
        sett.add(local+"@"+domain)
    }
    return sett.size
};

File: 965-unique-email-addresses/unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        s = set()
        for email in emails:
            e = email.split("@")
            local = e[0].split("+")[0]
            domain = e[1]
            if "." in local:
                local = "".join([i for i in local if i != "."])
            s.add(local+"@"+domain)
        return len(s)